### DISCUSSION TITLE
bring dim into csharp.

### DISCUSSION THREAD

**Discussion by joserubenh** [dotnet/csharplang#5092] [Language Ideas] [answered]
I understand the long debate between var and a field specification. I get it.
However, why not use also the dim = sytax ... as in 

dim t = new List<string>();

I already have a Roslyn tool that detects the "dim" and changes to whatever type is on the right side, so I only have to write it down in visual studio, it would be great if that was done automatically. I've read all the reasons why not to, but at this point, I think is just stubborness.

**Comment by theunrepentantgeek** [comment]
Not stubborness. [-100 points](https://docs.microsoft.com/en-us/archive/blogs/ericgu/minus-100-points).

There's nothing better (or worse) about **dim** than about **var** - changing the keyword doesn't address the underlying issues at all.

Inferring the type of a field definition is something the C# LDM doesn't want, because it makes the public interface of the type accidental.

**Reply by IanKemp** [reply]
Does the -100 points rule still apply in this era in which number of upvotes on UserVoice and/or GitHub seems to determine which features the team decides to implement?

**Reply by HaloFour** [reply]
> Does the -100 points rule still apply in this era in which number of upvotes on UserVoice and/or GitHub seems to determine which features the team decides to implement?

The upvotes are considered input but the team is always final arbiter as to which features will be implemented and whether they are worth the effort (represented by the -100 points).

**Comment by joserubenh** [comment]
Every function in .NET returns a strongly-typed result. Why not pass that type to the compiler?
You could treat the "dim" (or "var")  a TypeParameter token that resolves to the type  returned by the function on the assigment.

This would not have the exact same behaviour as dim, dim should always be assigned at the declaration.

You write:     `dim somevar = new List<string>();`

When the code is being parsed by Roslyn dim, inheriting from TypeParameter would simply return the TypeParameter on the assignment, in the end, the compiler reads:  

 ` List<string>() somevar = new List<string>();`

new List<string>() will always return a type of List<string>, in what scenario the declaration could that be accidental?
you could even restrict the use of "dim" for constructors only.

I read this article by Eric Lippert https://ericlippert.com/2009/01/26/why-no-var-on-fields/ . I do understand the issue with the parsing stages and the fact that the type must be known before the type of a method can be identified.... but if var is restricted to  calling a constructor only, I don't see the issue.

**Reply by huoyaoyuan** [reply]
> is restricted to calling a constructor only

The solution for this is target-typed new. 
```cs
List<SomeType> field = new();
```

**Reply by joserubenh** [reply]
Ahhh... didn't see that. great!. Thanks.

**Comment by joserubenh** [comment, marked as answer]
The solution for this is target-typed new.
As pointed out by @huoyaoyuan 

List<SomeType> field = new();
